src/transaction.js

```diff
@@ -1,536 +1,482 @@
-    );
-    // https://github.com/bitcoin/bitcoin/blob/master/src/test/sighash_tests.cpp#L29
-    if (inIndex >= this.ins.length) return ONE;
-    // ignore OP_CODESEPARATOR
-    const ourScript = bscript.compile(
-      bscript.decompile(prevOutScript).filter(x => {
-        return x !== script_1.OPS.OP_CODESEPARATOR;
-      }),
-    );
-    const txTmp = this.clone();
-    // SIGHASH_NONE: ignore all outputs? (wildcard payee)
-    if ((hashType & 0x1f) === Transaction.SIGHASH_NONE) {
-      txTmp.outs = [];
-      // ignore sequence numbers (except at inIndex)
-      txTmp.ins.forEach((input, i) => {
-        if (i === inIndex) return;
-        input.sequence = 0;
-      });
-      // SIGHASH_SINGLE: ignore all outputs, except at the same index?
-    } else if ((hashType & 0x1f) === Transaction.SIGHASH_SINGLE) {
-      // https://github.com/bitcoin/bitcoin/blob/master/src/test/sighash_tests.cpp#L60
-      if (inIndex >= this.outs.length) return ONE;
-      // truncate outputs after
-      txTmp.outs.length = inIndex + 1;
-      // "blank" outputs before
-      for (let i = 0; i < inIndex; i++) {
-        txTmp.outs[i] = BLANK_OUTPUT;
-      }
-      // ignore sequence numbers (except at inIndex)
-      txTmp.ins.forEach((input, y) => {
-        if (y === inIndex) return;
-        input.sequence = 0;
-      });
+    addOutput(scriptPubKey, value) {
+        typeforce(types.tuple(types.Buffer, types.Satoshi), arguments);
+        // Add the output and return the output's index
+        return (this.outs.push({
+            script: scriptPubKey,
+            value,
+        }) - 1);
     }
-    // SIGHASH_ANYONECANPAY: ignore inputs entirely?
-    if (hashType & Transaction.SIGHASH_ANYONECANPAY) {
-      txTmp.ins = [txTmp.ins[inIndex]];
-      txTmp.ins[0].script = ourScript;
-      // SIGHASH_ALL: only ignore input scripts
-    } else {
-      // "blank" others input scripts
-      txTmp.ins.forEach(input => {
-        input.script = EMPTY_BUFFER;
-      });
-      txTmp.ins[inIndex].script = ourScript;
+    hasWitnesses() {
+        return this.ins.some(x => {
+            return x.witness.length !== 0;
+        });
     }
-    // serialize and hash
-    const buffer = Buffer.allocUnsafe(txTmp.byteLength(false) + 4);
-    buffer.writeInt32LE(hashType, buffer.length - 4);
-    txTmp.__toBuffer(buffer, 0, false);
-    return bcrypto.hash256(buffer);
-  }
-  hashForWitnessV1(inIndex, prevOutScripts, values, hashType, leafHash, annex) {
-    // https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#common-signature-message
-    typeforce(
-      types.tuple(
-        types.UInt32,
-        typeforce.arrayOf(types.Buffer),
-        typeforce.arrayOf(types.Satoshi),
-        types.UInt32,
-      ),
-      arguments,
-    );
-    if (
-      values.length !== this.ins.length ||
-      prevOutScripts.length !== this.ins.length
-    ) {
-      throw new Error('Must supply prevout script and value for all inputs');
+    stripWitnesses() {
```

Comment:
@junderw I wasn't sure what to do with these changes. Should I just not commit the build changes?

Reply:
It might be a version diff.

Wipe node_modules and install deps using `npm ci` and it should be correct. (That's what CI does)

I think master has a newer version of the formatter and maybe your local node_modules still had that newer version (whereas CI will pull the older version).

Reply:
Then just run `npm run format && npm run build` and it should give you only the diff in your js similar to the ts, and a new function for the d.ts types file.